functions/src/update-user-metrics.ts

```diff
@@ -57,21 +57,21 @@ export async function updateUserMetrics() {
   const userUpdates = await batchedWaitAll(
     users.map((user) => async () => {
       const userContracts = contractsByCreator[user.id] ?? []
-      const currentBets = (
-        await firestore
-          .collectionGroup('bets')
-          .where('userId', '==', user.id)
-          .get()
-      ).docs.map((d) => d.data() as Bet)
-      const betsByContractId = groupBy(currentBets, (b) => b.contractId)
+      const unresolvedBetContracts = contracts.filter(
+        (c) => !c.isResolved && c.uniqueBettorIds?.includes(user.id)
+      )
```

Comment:
Actually, there's another tricky thing here.

We actually save changes in daily, weekly, and monthly profit in the docs of `contract-metrics`. Technically, those need to be updated until 30 days after the market resolves, when 30-day profit would be 0.

The quick fix here is to update all markets that haven't resolved or have been resolved within the last 30 days.

Reply:
Hmm. I see your point. I guess just applying the quick fix for now sounds OK.